src/main/java/io/jenkins/plugins/coverage/CoverageChecksPublisher.java

```diff
@@ -0,0 +1,115 @@
+package io.jenkins.plugins.coverage;
+
+import edu.hm.hafner.util.VisibleForTesting;
+import io.jenkins.plugins.checks.api.*;
+import io.jenkins.plugins.checks.api.ChecksOutput.ChecksOutputBuilder;
+import io.jenkins.plugins.checks.api.ChecksDetails.ChecksDetailsBuilder;
+import io.jenkins.plugins.coverage.targets.*;
+
+import java.util.*;
+
+/**
+ * Publishes coverage as checks to scm platforms.
+ *
+ * @author Kezhi Xiong
+ */
+class CoverageChecksPublisher {
+    private final CoverageAction action;
+
+    CoverageChecksPublisher(final CoverageAction action) {
+        this.action = action;
+    }
+
+    void publishChecks() {
+        ChecksPublisher publisher = ChecksPublisherFactory.fromRun(action.getOwner());
+        publisher.publish(extractChecksDetails());
+    }
+
+    @VisibleForTesting
+    ChecksDetails extractChecksDetails() {
+        CoverageResult result = action.getResult();
+        ChecksOutput output = new ChecksOutputBuilder()
+                .withTitle(extractChecksTitle(result))
+                .withSummary("")
+                .withText(extractChecksText(result))
+                .build();
+
+        return new ChecksDetailsBuilder()
+                .withName("Code Coverage")
+                .withStatus(ChecksStatus.COMPLETED)
+                .withConclusion(ChecksConclusion.SUCCESS)
+                .withDetailsURL(action.getAbsoluteUrl())
+                .withOutput(output)
+                .build();
+    }
+
+    private String extractChecksText(final CoverageResult result) {
+        Map<CoverageElement, Ratio> ratios = result.getResults();
+        Map<CoverageElement, Ratio> lastRatios = getLastRatios(result);
```

Comment:
I wonder if it would make sense to separate the total report and the delta report? Maybe we can discuss that topic in our meeting today. This question is also relevant for the analysis report (new and total warnings).